lib/multiplexer/multiplexer.go

```diff
@@ -234,7 +251,7 @@ func (m *Mux) detectAndForward(conn net.Conn) {
 	connWrapper, err := m.detect(conn)
 	if err != nil {
 		if trace.Unwrap(err) != io.EOF {
-			m.Warning(trace.DebugReport(err))
+			m.logLimiter.Log(m.Entry, log.WarnLevel, trace.DebugReport(err))
```

Comment:
I'm not quite understanding why this works. Does entry get set every time `mux.Entry.WithError(err).Error("Mux encountered err serving")` gets called in lib/service/service.go? And `detect` will always return an error when that happens?

Reply:
From what I can tell `mux.Entry.WithError` doesn't mutate `mux.Entry`, it returns a new one.